src/go/protos/protos.go

```diff
@@ -0,0 +1,18 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protos
+
+//go:generate go install google.golang.org/protobuf/cmd/protoc-gen-go
```

Comment:
I've not used `go generate` - are these commented out lines actually acted upon by go tooling? Or are they here to describe what needs to be executed by a shell script / Makefile / by hand for notekeeping purposes?

Reply:
these aren't comments, they're instructions for go generate
https://go.dev/blog/generate

Reply:
https://go.dev/blog/generate

yeah, this is a `go` tool built-in that does simple one-liners to re-gen code since that's the preference of the original golang authors.

Reply:
maybe it would be good to follow the same system we use in orc8r. It will save time and will make the system homogeneous between gateways.

We trigger generation of all protos with `./build.py -g` at `~/magma/orc8r/cloud/docker`. That command runs the following steps 
- Creates a build pod and run a command [here](http://github.com/magma/magma/blob/cdc9dd678ccb35e9dd24c8b9a4aefc2894444dba/orc8r/cloud/docker/build.py#L88-L88) 
- Then we run `make fullgen` command inside the a pod we use for building or8cr [here](http://github.com/magma/magma/blob/cdc9dd678ccb35e9dd24c8b9a4aefc2894444dba/orc8r/cloud/Makefile#L117-L117)
- That command will run a `make gen` on all the `cloud/go` folders in our system. For example for feg is [this](http://github.com/magma/magma/blob/cdc9dd678ccb35e9dd24c8b9a4aefc2894444dba/feg/cloud/go/Makefile#L2-L2)
- which finaly runs [gp generate ./...](http://github.com/magma/magma/blob/cdc9dd678ccb35e9dd24c8b9a4aefc2894444dba/orc8r/cloud/go/module.mk#L32-L32)
- The protos ends up under the cloud/go/proto folder

That command generates all the protos used in magma for go. FEG feeds from those. So if you create a `cloud/go` folder in AGW We could do the same for AGW and centralize.

Note that protos are shared between agw feg and orc8c so it makes sense to have one single place to generates then than 3. 

Note that this command also generates swagger modules. So once we have swagger on AGW that will come for free

Reply:
a few things to note here:

1. we're trying hard to not have local go.mod rewrites that go outside of `src/go`
2. generate/build/etc. should be fast, even from scratch. with go, there's no reason to need a VM/docker build appliance; developers should be able to do most of their development in their native environment if desired (macos, linux, windows, etc.). this yields the best/fastest dev loops and reduces "dev infra" setup/maintenance time
3. the go namespace directives in our proto files appear to be all fubar (although this would be more a point towards moving everything to src/go instead of spread out across <box>/<area>/<deployment env>/<programming language> directories)

and also, 4. I think this entire area is temporary while we figure out bazel :)